Removal of items.otb format and sid/cid situation

Tofame opened the pull request:
**Prelude**
``.otb`` format is deemed to not be implementing anything new, it's just a bridge for a unnecessary convention of client-id and server-id, which was allegedly created because of some weird item removal YEARS ago.
My personal suspicision is, that for edit of ``.dat`` a tool allowing removal of items was possible for example from the middle, contrary to the Object Builder tool which only allows to remove from the end. That could justify why there is more server ids total than client ids, and why they are connected in such manner.

**PR features**
- removal of ``items.otb``, they are no longer needed, everything is loaded from ``.dat``
- removal of `ClientIdToServerIdMap` class, which was cid-sid bridge, no longer needed.
- ``itemsDatPath`` config variable, so by default `.dat` is searched in ``items/dat``, but as QoL, users can point it to e.g. client, for ease of local development and faster updates, or whatever. Was quite unsure, if it's really crucial, but I saw the path - 'data/items/items.otb' used in 2 different places, and thought that config could atleast keep it more concise too.

**Explanation on `.dat`**
- This .dat I obtained by using a default one and using it through my custom [app](https://github.com/Tofame/OtbDatTransformata)
- The custom .dat is achieved by loading both `.otb` and `.dat` at once, and giving in an output a `.dat` file that is ordered by 'server ids'.

**What's more required to be done**
- Adaptation of RME code, maybe someone would be willing 🗡️ 
- Testing if I parsed everything properly... just walk around, click on things, read/write signs, books etc.

Codinablack commented on src/items.h:
```diff
@@ -384,7 +384,7 @@ class ItemType
 		bool rotatable = false;
 		bool useable = false;
-		bool moveable = false;
+		bool moveable = true;
```
The last time we made a change like this it had some un-forseen consequences. Are you sure you have covered all situations that are affected by doing this?

Tofame replied:
It's on my todo to check this, good catch, the reasoning behind this is mostly:
<img width="256" height="69" alt="obraz" src="https://github.com/user-attachments/assets/d8f31d9d-e6fa-4a89-aaf0-df51704402f7" />
On the left being item editor (otb) and on the right object builder (dat). They as can be seen have confusing disparity.
What needs to be checked is what finally is stored in the .dat. As long as it's really unmovable, then yes, the default `moveable` should be `true`, since OB is responsible for setting it for false (if confirmed to be true).

If Item Editor really has it inverted, then it makes sense it was by default `false`, as the reason would be that item editor was responsible for making it true.

Tofame replied:
Yes, so basically, I'm doing what item editor is - it inverts - has default moveable = true, and if detects Unmovable turns it to false, and saves it to OTB. Then TFS was reading the **movable** flag and setting things to true.

I, am, now skipping OTB part, so ye, reading unmovable.

Codinablack commented:
Overall I think this commit looks pretty solid! I did some digging around to be sure that the flags were not being called on directly somewhere, and that was not the case. The enums don't follow the typical case convention, but honestly I have been really leaning towards doing the enums the way you have them anyways, for all enums. That's how I do them in my other project, and so I'm not against mixing the cases for now... I will probably convert all the old ones sometime anyways.

I really liked the use of fmt's formatting options, and the fact that you built the code as if it's ready at any moment to be tapped into, in those places we are not utilizing now. 

I didn't see any obvious problems, red fags, or causes for concern. LGTM!

javiertringol commented:
A question from my ignorance:

With this change, will all the `serverId` need to be changed to `clientId`?
current code
Example:
```lua 
local action = Action()
function action.onUse(player, item, fromPos, target, toPos, isHotkey)
--me code...
action:id(ItemServerId -> ItemClientId) -- change here
action:register()
```

Or:

Do these `itemids `also need to be changed from `serverId `to `clientId`?

```xml
<movevent event="DeEquip" itemid="2212" slot="ring" function="onDeEquipItem" />
<movevent event="Equip" itemid="2213" slot="ring" function="onEquipItem" />
```

Tofame commented:
@javiertringol no, every `id` used in either lua scripts or c++ sources is still working in the server. There is no need to change anything 👍

The only likely change (that may be only applying to a few people anyway), if someone was creating a custom module, with e.g. `.otui` language, and they used hardcoded client ids there, then they will need to update those.
So this is only likely affecting **OTC + Custom modules**

Tofame commented:
Screen of me standing next to custom added item, added through rme on the map (not thrown by me on the ground).
So all in all this screen confirms that:
- Rme works
- Otc works
- Object builder works
- Item Editor also works (.otb is still needed for now to be included in rme and updated)
<img width="294" height="291" alt="obraz" src="https://github.com/user-attachments/assets/7dcccf4a-ead1-4de0-a9eb-b5ecbe487a1d" />
